File: 프로그래머스/2/12951. JadenCase 문자열 만들기/JadenCase 문자열 만들기.java
class Solution {
    public String solution(String s) {
        StringBuilder sb = new StringBuilder();
        boolean isStart = true; // 단어의 시작 여부

        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);

            if (c == ' ') {
                sb.append(c);
                isStart = true;
            } else {
                if (isStart) {
                    sb.append(Character.toUpperCase(c));
                    isStart = false;
                } else {
                    sb.append(Character.toLowerCase(c));
                }
            }
        }
        return sb.toString();
    }
}

File: 프로그래머스/2/12951. JadenCase 문자열 만들기/JadenCase 문자열 만들기.js
function solution(s) {
    let answer = '';
    let isStart = true; // 단어 시작 여부

    for (let i = 0; i < s.length; i++) {
        const c = s[i];

        if (c === ' ') {
            answer += c;
            isStart = true;
        } else {
            if (isStart) {
                answer += c.toUpperCase();
                isStart = false;
            } else {
                answer += c.toLowerCase();
            }
        }
    }
    return answer;
}
